recipes/google-cloud-cpp/all/conanfile.py

```diff
@@ -0,0 +1,279 @@
+            raise ConanInvalidConfiguration("Fails to compile for Windows as a DLL")
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+
+        if self.settings.compiler == 'gcc' and tools.Version(self.settings.compiler.version) < "5.4":
+            raise ConanInvalidConfiguration("Building requires GCC >= 5.4")
+        if self.settings.compiler == 'clang' and tools.Version(self.settings.compiler.version) < "3.8":
+            raise ConanInvalidConfiguration("Building requires clang >= 3.8")
+        if self.settings.compiler == 'Visual Studio' and tools.Version(self.settings.compiler.version) < "16":
+            raise ConanInvalidConfiguration("Building requires VS >= 2019")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def requirements(self):
+        self.requires('protobuf/3.15.5')
+        self.requires('grpc/1.37.1')
+        self.requires('nlohmann_json/3.9.1')
+        self.requires('crc32c/1.1.1')
+        self.requires('abseil/20210324.0')
+        self.requires('libcurl/7.75.0')
+        self.requires('openssl/1.1.1k')
+        # TODO: Add googleapis once it is available in CCI (now it is embedded)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = 0
+
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK"] = False
+
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_BIGTABLE"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_BIGQUERY"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_SPANNER"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_STORAGE"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_FIRESTORE"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_PUBSUB"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_IAM"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_LOGGING"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_GENERATOR"] = True
+
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, 'lib', "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+
+    def package_info(self):
+        # Using componen google-cloud-common, so it reduces the risk of conflict (pkgconfig generator)
```

Comment:
Looking at installed pkgconfig files, pkg_config files are not properly emulated. This component `google-cloud-common` could have been called `common` with pkgconfig name `google_cloud_cpp_common`.

CMake targets seem ok, but generators can't emulate official config files, and google-cloud-cpp generates more than one config file.
Actually, there is no `google-cloud-cpp-config.cmake` file upstream.

Reply:
I find your suggestion much more convenient. From the consumer perspective, I prefer to use `google-cloud-cpp::common` over `google-cloud-cpp::google-cloud-common` to refer to that component. I think it is worth a PR and probably a suggestion to follow for other PRs, wdyt @prince-chrismc ?

Reply:
From the CMake perspective `google-cloud-cpp::common` should be working since we added those names 🤔 both names are **currently** available AFAIK.

My original comment stems for all the times Space flagged this potential issue, I did not check the pkg_config log but we could have done the inverse as is suggested. So I have no objections here.

Instead of changing the component name we could simple just add the corrected name? Avoid changing it's usage in the recipe

```suggestion
self.cpp_info.components["google-cloud-common"].names["pkg_config"] = "google_cloud_cpp_common"
```

I assume they did not use "common" since it was too common xD

Reply:
My comment was about Conan components, not related to CMake targets. From a consumer package you will refer to that component as `google-cloud-cpp::google-cloud-common` (now). I'm voting to changing it back to `google-cloud-cpp::common`.

CMake targets are not changed. Pkgconfig would require the line you suggested.